pkg/persistence/account/writer/writer.go

```diff
@@ -145,15 +156,27 @@ func toPersistenceGroups(groups map[string]account.Group) []persistence.Group {
 				Permissions: e.Permissions,
 				Policies:    transformRefs(e.Policies),
 			}
+			// sort permissions so that they are stable within a persisted file
+			slices.SortFunc(envs[i].Permissions, caseInsensitiveLexicographicSmaller)
 		}
+		// sort envs by name so that they are stable within a persisted file
+		slices.SortFunc(envs, func(a, b persistence.Environment) bool {
+			return caseInsensitiveLexicographicSmaller(a.Name, b.Name)
+		})
 		mzs := make([]persistence.ManagementZone, len(v.ManagementZone))
 		for i, e := range v.ManagementZone {
 			mzs[i] = persistence.ManagementZone{
 				Environment:    e.Environment,
 				ManagementZone: e.ManagementZone,
 				Permissions:    e.Permissions,
 			}
+			// sort permissions so that they are stable within a persisted file
+			slices.SortFunc(mzs[i].Permissions, caseInsensitiveLexicographicSmaller)
 		}
+		// sort mzs by env and name so that they are stable within a persisted file
+		slices.SortFunc(mzs, func(a, b persistence.ManagementZone) bool {
+			return strings.ToLower(a.Environment) <= strings.ToLower(b.Environment) && caseInsensitiveLexicographicSmaller(a.ManagementZone, b.ManagementZone)
```

Comment:
why not use the compare func for Environment as well?

Reply:
I want MZ for the same environment to be sorted by MZ under that environment. the compare func only does 'smaller' not equal. 
Maybe I'm thinking too complicated though?

Reply:
i mean why doing it inline here, whereas for the other's compare you've created a function.

Reply:
anyway, it's a nit

Reply:
Ah.. I created a function because I did the same check in several places. In this case I only do it once, so other than documentation via naming I saw/see no reason for adding a function here.